protocol-units/execution/opt-executor/src/executor/transaction_pipe.rs

```diff
@@ -43,68 +39,63 @@ impl Executor {
 			match request {
 				MempoolClientRequest::SubmitTransaction(transaction, callback) => {
-					// Pre-execute Tx to validate its content.
-					// Re-create the validator for each Tx because it uses a frozen version of the ledger.
-					// let vm_validator = VMValidator::new(Arc::clone(&self.db.reader));
-					// let tx_result = vm_validator.validate_transaction(transaction.clone())?;
-
-					// let status = if let Some(vm_status) = tx_result.status() {
-					// 	// If the verification failed, return the error status.
-					// 	let ms = MempoolStatus::new(MempoolStatusCode::VmError);
-					// 	(ms, Some(vm_status))
-					// } else {
+
+					// Shed load.
+					// Low-ball the load shedding for now with 4096 transactions allowed in flight.
+					// For now, we are going to consider a transaction in flight until it exits the mempool and is sent to the DA as is indicated by WriteBatch.
+					if self.transactions_in_flight.load(std::sync::atomic::Ordering::Relaxed) > 2^12 {
+						info!("Transaction ins flight: {:?}, shedding load", self.transactions_in_flight.load(std::sync::atomic::Ordering::Relaxed));
+						let status = MempoolStatus::new(MempoolStatusCode::MempoolIsFull);
+						callback.send(Ok((status.clone(), None))).map_err(
+							|e| TransactionPipeError::InternalError(format!("Error sending transaction: {:?}", e))
+						)?;
+						return Ok(())
+					}
+
 					let status = {
 						// add to the mempool
-						{
-							let mut core_mempool = self.core_mempool.write().await;
-
-							debug!(
-								"Adding transaction to mempool: {:?} {:?}",
-								transaction,
-								transaction.sequence_number()
-							);
-							let status = core_mempool.add_txn(
-								transaction.clone(),
-								0,
-								transaction.sequence_number(),
-								TimelineState::NonQualified,
-								true,
-							);
-
-							match status.code {
-								MempoolStatusCode::Accepted => {
-									debug!("Transaction accepted: {:?}", transaction);
-								}
-								_ => {
-									debug!("Transaction not accepted: {:?}", status);
-									Err(TransactionPipeError::TransactionNotAccepted(status))?;
-								}
-							}
-						}
-
-						// send along to the receiver
-						transaction_channel
+						let mut core_mempool = self.core_mempool.write().await;
+
+						debug!(
+							"Adding transaction to mempool: {:?} {:?}",
+							transaction,
+							transaction.sequence_number()
+						);
+						core_mempool.add_txn(
+							transaction.clone(),
+							0,
+							transaction.sequence_number(),
+							TimelineState::NonQualified,
+							true,
+						)
+					};
+
+					// increment 
+					match &status.code {
+						MempoolStatusCode::Accepted => {
+							// send along to the receiver
+							transaction_channel
 							.send(transaction)
 							.await
 							.map_err(|e| anyhow::anyhow!("Error sending transaction: {:?}", e))?;
+							self.transactions_in_flight.fetch_add(1, std::sync::atomic::Ordering::Relaxed);
+						},
+						_ => {}
```

Comment:
Silently doing nothing here.

If this case is never supposed to happen, perhaps at least put out a `warn!()` tracing event in case it does?

Reply:
Any comment on this? Is it a legitimate case to silently ignore?